lib/pbench/server/api/resources/users_api.py

```diff
@@ -390,31 +403,44 @@ def put(self, target_username):
                         "message": "failure message"
                     }
         """
-        post_data = request.get_json()
-        if not post_data:
+        user_payload = request.get_json()
+        if not user_payload:
             self.logger.warning("Invalid json object: {}", request.url)
             abort(HTTPStatus.BAD_REQUEST, message="Invalid json object in request")
 
         try:
-            target_user = self.operation_permitted(target_username)
-            if not target_user:
+            target_user = self.get_valid_target_user(target_username)
+        except Exception:
+            self.logger.exception(
+                "Exception occurred in PUT request from user {!r} while validating target user {!r}",
+                self.auth.token_auth.current_user().username,
+                target_username,
+            )
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+        if not target_user:
+            if self.auth.token_auth.current_user().is_admin():
                 self.logger.warning(
-                    "User {} is not authorized to delete user {}.",
+                    "User {} performed update operation but user {} is not found.",
                     self.auth.token_auth.current_user().username,
                     target_username,
                 )
                 abort(
-                    HTTPStatus.FORBIDDEN,
-                    message=f"Not authorized to update information about user {target_username}",
+                    HTTPStatus.NOT_FOUND, message=f"User {target_username} not found",
                 )
-        except Exception:
-            self.logger.exception("Exception occurred while verifying the user")
-            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+            self.logger.warning(
+                "User {} is not authorized to modify user {}.",
+                self.auth.token_auth.current_user().username,
+                target_username,
+            )
+            abort(
+                HTTPStatus.FORBIDDEN,
+                message=f"Not authorized to update information about user {target_username}",
+            )
```

Comment:
I think we have three copies of this code which are all the same except for the request type (`GET`, `PUT`, `DELETE`).  It would be nice to have them using common code, but I feel like we shouldn't be burying calls to `abort()` inside subroutines.  Does anyone else have opinions on this?

I suppose we could have a helper routine which returns a target user or a status/message pair (the details of the mechanism for that are TBD) so that this function could receive either the information that it wants or the arguments needed for the appropriate `abort()` call, but perhaps that's overdoing it.  Thoughts?

Reply:
Good point, helper function `get_valid_target_user` could return a target_user along with the status code that can be used to proceed with a requested operation, unless some other exception occurs in which case we would be returning 50X status

Reply:
I'm kind of fond of the clean simplicity of the current implementation of `get_valid_target_user()`, so, if you're up for this, @npalaska, rather than enhancing `get_valid_target_user()`, I would suggest implementing a wrapper for it.

Reply:
OK, I see that there is an "economy of scale" to be gained by integrating the change into `get_valid_target_user()`, so nevermind -- it's good as it is now.